builtin/logical/pki/acme_state.go

```diff
@@ -165,44 +177,58 @@ func (a *acmeState) UpdateAccount(ac *acmeContext, acct *acmeAccount) error {
 	return nil
 }
 
-func cleanKid(keyID string) string {
-	pieces := strings.Split(keyID, "/")
-	return pieces[len(pieces)-1]
-}
-
 func (a *acmeState) LoadAccount(ac *acmeContext, keyID string) (*acmeAccount, error) {
-	kid := cleanKid(keyID)
-
-	entry, err := ac.sc.Storage.Get(ac.sc.Context, acmeAccountPrefix+kid)
+	entry, err := ac.sc.Storage.Get(ac.sc.Context, acmeAccountPrefix+keyID)
 	if err != nil {
 		return nil, fmt.Errorf("error loading account: %w", err)
 	}
 	if entry == nil {
-		return nil, fmt.Errorf("account not found: %w", ErrMalformed)
+		return nil, nil
```

Comment:
Remind me again why we did this? :laughing:  Other than returning `ErrAccountDoesNotExist` here rather than `ErrMalformed` in some scenarios (and perhaps returning `ErrAccountDoesNotExist` is most correct anyways), was there a benefit?

It seems like we'd never want to partially load an account and assume we have one when we don't actually...

Reply:
Well, when I added `LoadAccountByKey` I had scenarios that I didn't want to error out completely if we couldn't load the account. I felt it odd that we would have very different behavior across `LoadAccount` and `LoadAccountByKey` so I changed it.

Would you rather keep the ErrAccountDoesNotExist and the callers check for that instead of nil? It was a bit of trying to keep things a little consistent in our inconsistency within the plugin for dealing with missing things. 🤷 

Or would you rather see a different return behavior around `LoadAccount` and `LoadAccountByKey`? As you say `LoadAccount` probably doesn't have any real use cases to not error out completely if it fails to find something.

Reply:
Eh, nah, its fine. I think there's only one scenario where you'd want to discard this error ever, which is when you're using `LoadAccountByKey` to see if the key exists in the creation handler. Here, I think explicitly ignoring the error once is cleaner than having to remember every time to guard against it not existing...? My 2c though :-)